### PR TITLE
MDS Volume statstics

### PR DESCRIPTION
**Pull request by gzagatti**
### I have:

- [x] Formatted any Python files with [black](https://github.com/ambv/black)
- [x] Brought the branch up to date with master
- [x] Added any relevant Github labels
- [x] Added tests for any new additions
- [x] Added or updated any relevant documentation
- [ ] Added an Architectural Decision Record (ADR), if appropriate
- [x] Added an [MPLv2 License Header](https://www.mozilla.org/en-US/MPL/headers/) if appropriate
- [ ] Updated the [Changelog](https://github.com/Flowminder/FlowKit/blob/master/CHANGELOG.md) 

### Description

This PR adds MDS Volume statistics per subscriber.

### CODE REVIEW

File: flowmachine/flowmachine/features/subscriber/mds_volume.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    hours : 2-tuple of floats, default 'all'
+        Restrict the analysis to only a certain set
+        of hours within each day.
+    subscriber_identifier : {'msisdn', 'imei'}, default 'msisdn'
+        Either msisdn, or imei, the column that identifies the subscriber.
+    subscriber_subset : str, list, flowmachine.core.Query, flowmachine.core.Table, default None
+        If provided, string or list of string which are msisdn or imeis to limit
+        results to; or, a query or table which has a column with a name matching
+        subscriber_identifier (typically, msisdn), to limit results to.
+    tables : str or list of strings, default 'events.mds'
```
Comment (greenape) [inline flowmachine/flowmachine/features/subscriber/mds_volume.py]: Does it make sense to have this param, given it explicitly only deals with MDS?

File: flowmachine/flowmachine/features/subscriber/mds_volume.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+        if self.statistic not in valid_stats:
+            raise ValueError(
+                "{} is not a valid statistic. Use one of {}".format(
+                    self.statistic, valid_stats
+                )
+            )
+
+        if volume not in {"total", "upload", "download"}:
+            raise ValueError("{} is not a valid volume.".format(self.volume))
```
Comment (greenape) [inline flowmachine/flowmachine/features/subscriber/mds_volume.py]: fstring.

File: flowmachine/flowmachine/features/subscriber/mds_volume.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+from ..utilities.sets import EventsTablesUnion
+from .metaclasses import SubscriberFeature
+
+valid_stats = {"count", "sum", "avg", "max", "min", "median", "stddev", "variance"}
+
+
+class MDSVolume(SubscriberFeature):
+    """
+    This class calculates statistics associated with MDS volume usage.
```
Comment (greenape) [inline flowmachine/flowmachine/features/subscriber/mds_volume.py]:
```suggestion
    This class calculates statistics associated with MDS (mobile data session) volume usage.
```

File: flowmachine/flowmachine/features/subscriber/mds_volume.py (hunk 1 above)
Comment (gzagatti) [reply flowmachine/flowmachine/features/subscriber/mds_volume.py]: I think this was for consistency sake. You never know if someone decides to create a curated events table with mds data.

**Comment by gzagatti** [comment]
Do you want to open an issue for "there should really be a more generic "calculate stats over all the columns" type class that these kind of ones can exploit"? I agree with you.

**Pull request merged**